lib/portlayer/event/vsphere/manager.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vsphere
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"reflect"
+
+	"github.com/vmware/vic/lib/portlayer/event"
+	"github.com/vmware/vic/pkg/vsphere/session"
+
+	vmwEvents "github.com/vmware/govmomi/event"
+	"github.com/vmware/govmomi/vim25/types"
+
+	"golang.org/x/net/context"
+)
+
+type vSphereManager struct {
+	vmwManager *vmwEvents.Manager
+	callbacks  map[string]func(event.Event, *session.Session)
+	blacklist  map[types.ManagedObjectReference]string
+	objects    map[string]types.ManagedObjectReference
+	sess       *session.Session
+}
+
+func NewEventManager(sess *session.Session) *vSphereManager {
+	return &vSphereManager{
+		callbacks: make(map[string]func(event.Event, *session.Session)),
+		blacklist: make(map[types.ManagedObjectReference]string),
+		objects:   make(map[string]types.ManagedObjectReference),
+		sess:      sess,
+	}
+
+}
+
+func (mgr *vSphereManager) AddMonitoredObject(ref string) error {
+	log.Debugf("EventManager added (%s) to monitor", ref)
+	moRef := types.ManagedObjectReference{}
+	if !moRef.FromString(ref) {
+		return fmt.Errorf("Invalid Managed Object provided for Montioring(%s)", ref)
+	}
+	mgr.objects[ref] = moRef
+	return nil
+}
+
+func (mgr *vSphereManager) RemoveMonitoredObject(ref string) {
+	delete(mgr.objects, ref)
+}
+
+func (mgr *vSphereManager) monitoredObjects() []types.ManagedObjectReference {
+	refs := make([]types.ManagedObjectReference, 0, len(mgr.objects))
+	for k := range mgr.objects {
+		refs = append(refs, mgr.objects[k])
+	}
+	return refs
+}
+
+// blacklist object from callbacks
```

Comment:
vet will whine (and our default script will ignore) because comments are starting with lower case letters. Can you match with fn names?